src/server/socketserver.cpp

```diff
@@ -9,33 +9,45 @@ SocketServer::~SocketServer() {
     stop();
 }
 
-void SocketServer::setupSocket() {
-    sock = getSocket();
-    if (sock == -1) {
-        std::cerr << "socket error: " << strerror(errno) << "\n";
-        return;
+int SocketServer::setupSocket() {
+
+    int result = 0;
+
+    this->sock = getSocket();
+
+    if (this->sock == -1) {
+        std::cerr << "socket error: " << strerror(errno) << std::endl;
+        result = -1;
```

Comment:
I think it would be better to handle this with exceptions, at least that appears to be more in line with C++ paradigms for me.

Also, when handling this as a result, it's probably easier to just `return -1` here to avoid the need to check the value in the rest of the method.